### PR TITLE
[multi-DUT] making test_interfaces multi-DUT ready

### PR DESCRIPTION
**Pull request by yxieca**
### Description of PR
Summary:
making test_interfaces multi-DUT ready

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
making test_interfaces multi-DUT ready

#### How did you do it?
- enumerate DUTs of testbed to run test.
- improve asserts.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

#### How did you verify/test it?
Run test on a multi-DUT testbed.

platform linux2 -- Python 2.7.12, pytest-4.6.5, py-1.9.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /var/src/sonic-mgmt/tests, inifile: pytest.ini
plugins: ansible-2.2.2, forked-1.3.0, xdist-1.28.0, html-1.22.1, repeat-0.8.0, metadata-1.10.0
collected 2 items                                                                                                                                                                        

test_interfaces.py::test_interfaces[str2-7050cx3-acs-06] PASSED                                                                                                                    [ 50%]
test_interfaces.py::test_interfaces[str2-7050cx3-acs-07] PASSED                                                                                                                    [100%]

### CODE REVIEW

File: tests/test_interfaces.py [hunk 1]
```diff
@@ -18,7 +21,8 @@ def test_interfaces(duthost):
         # verify no ipv4 address for each port channel member
         for member in v['members']:
             ans_ifname = "ansible_%s" % member
-            assert not host_facts[ans_ifname].has_key("ipv4")
+            pytest_assert("ipv4" not in host_facts[ans_ifname],
+                          "LAG member {} has IP address {}".format(ans_ifname, host_facts[ans_ifname]))
```
Comment (neethajohn) [inline tests/test_interfaces.py:25]: for ip addr - host_facts[ans_ifname]["ipv4"]

File: tests/test_interfaces.py [hunk 2]
```diff
@@ -27,7 +31,8 @@ def test_interfaces(duthost):
         # verify no ipv4 address for each vlan member
         for member in v['members']:
             ans_ifname = "ansible_%s" % member
-            assert not host_facts[ans_ifname].has_key("ipv4")
+            pytest_assert("ipv4" not in host_facts[ans_ifname],
+                          "vlan member {} has IP address {}".format(ans_ifname, host_facts[ans_ifname]))
```
Comment (neethajohn) [inline tests/test_interfaces.py:35]: for ip addr - host_facts[ans_ifname]["ipv4"]

File: tests/test_interfaces.py (hunk 1 above)
Comment (yxieca) [reply tests/test_interfaces.py:25]: I am leaning towards show a bit more information than just ipv4.

Comment (yxieca) [reply tests/test_interfaces.py]: Changed according to your suggestion. The benefit is that the error message is more accurate.

File: tests/test_interfaces.py (hunk 2 above)
Comment (yxieca) [reply tests/test_interfaces.py]: Sure. done.

File: tests/test_interfaces.py (hunk 1 above)
Comment (yxieca) [reply tests/test_interfaces.py:25]: Sorry, I backed out the change. The message was calculated regardless of the assertion condition, so there will be key error in success cases.

File: tests/test_interfaces.py (hunk 2 above)
Comment (yxieca) [reply tests/test_interfaces.py:35]: Please see above. There could be key error if we do so.

File: tests/test_interfaces.py (hunk 1 above)
Comment (neethajohn) [reply tests/test_interfaces.py:25]: oh yes. my bad

**Review by neethajohn — APPROVED** [review]

**Comment by yxieca** [comment]
retest vsimage please

**Comment by yxieca** [comment]
retest vsimage please

**Pull request merged**